storage/src/test/java/org/apache/kafka/tiered/storage/integration/TransactionsWithTieredStoreTest.java

```diff
@@ -0,0 +1,135 @@
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public void modifyConfigs(Seq<Properties> props) {
+        for (Properties p : JavaConverters.seqAsJavaList(props)) {
+            p.putAll(overridingProps());
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public Seq<Properties> kraftControllerConfigs() {
+        return JavaConverters.asScalaBuffer(Collections.singletonList(overridingProps())).toSeq();
+    }
+
+    private Properties overridingProps() {
+        int numRemoteLogMetadataPartitions = 3;
+        boolean deleteOnClose = false;
+        return createPropsForRemoteStorage(testClassName, storageDirPath, brokerCount(),
+                numRemoteLogMetadataPartitions, deleteOnClose);
+    }
+
+    @Override
+    public Properties topicConfig() {
+        boolean enableRemoteStorage = true;
+        long localLogRetentionBytes = 1;
+        int maxBatchCountPerSegment = 1;
+        Properties overridingTopicProps = super.topicConfig();
+        overridingTopicProps.putAll(createTopicConfigForRemoteStorage(
+                enableRemoteStorage, localLogRetentionBytes, maxBatchCountPerSegment));
+        return overridingTopicProps;
+    }
+
+    // NOTE: Not able to refer TestInfoUtils#TestWithParameterizedQuorumName() in the ParameterizedTest name.
+    @ParameterizedTest(name = "{displayName}.quorum={0}")
+    @ValueSource(strings = {"zk", "kraft"})
+    public void testFencingOnTransactionExpiration(String quorum) {
+        // FIXME: Disabled the test with tiered storage since it's failing and needs to be fixed.
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public void maybeWaitForAtLeastOneSegmentUpload(scala.collection.immutable.Seq<TopicPartition> topicPartitions) {
+        JavaConverters.seqAsJavaList(topicPartitions).forEach(topicPartition -> {
+            List<BrokerLocalStorage> localStorages = JavaConverters.bufferAsJavaList(brokers()).stream()
+                    .map(b -> new BrokerLocalStorage(b.config().brokerId(), b.config().logDirs().head(), STORAGE_WAIT_TIMEOUT_SEC))
+                    .collect(Collectors.toList());
+            Optional<BrokerLocalStorage> brokerLocalStorageOpt = localStorages
+                    .stream()
+                    // Select brokers which are assigned a replica of the topic-partition
+                    .filter(s -> isAssignedReplica(topicPartition, s.getBrokerId()))
+                    // Filter out inactive brokers, which may still contain log segments we would expect
+                    // to be deleted based on the retention configuration.
+                    .filter(s -> isAlive(s.getBrokerId()))
+                    .findFirst();
+            brokerLocalStorageOpt.ifPresent(localStorage ->
+                    // Wait until the brokers local storage have been cleared from the inactive log segments.
+                    localStorage.waitForAtLeastEarliestLocalOffset(topicPartition, 1L));
+        });
+    }
```

Comment:
I'm a little confused here. From the test logic, I think we're expecting these topicPartitions should have at least 1 segment uploaded, right? So, it's not `maybe`. And then, with the current implementation, even if there's no segment get uploaded, the test is still passed. Is that expected?

Reply:
No, this is not expected. One or more segments needs to be uploaded by all the replicas of a partition. Instead of `findFirst` broker, we have to wait for all the brokers. Updated with the latest commit.

Reply:
The reason for using the prefix `maybe` is that in `TransactionsTest` without tiered storage, we won't be uploading any segments.